setup.py

```diff
@@ -58,342 +62,54 @@
-        get_pytorch_cmake_prefix,
-        get_pytorch_use_distributed,
-    )
-
-    # this is used to suppress import error.
-    # so we can get the right pytorch prefix for cmake
-    import logging
-
-    logger = logging.getLogger("nvfuser")
-    logger_level = logger.getEffectiveLevel()
-    logger.setLevel(logging.CRITICAL)
-
-    pytorch_cmake_config = "-DCMAKE_PREFIX_PATH=" + get_pytorch_cmake_prefix()
-
-    logger.setLevel(logger_level)
-
-    pytorch_use_distributed = get_pytorch_use_distributed()
-
-    # generate cmake directory
-    cmd_str = [
-        get_cmake_bin(),
-        pytorch_cmake_config,
-        "-DCMAKE_BUILD_TYPE=" + BUILD_TYPE,
-        f"-DCMAKE_INSTALL_PREFIX={install_prefix}",
-        f"-DNVFUSER_CPP_STANDARD={CPP_STANDARD}",
-        f"-DUSE_DISTRIBUTED={pytorch_use_distributed}",
-        "-B",
-        cmake_build_dir,
-    ]
-    if BUILD_WITH_UCC:
-        cmd_str.append("-DNVFUSER_STANDALONE_BUILD_WITH_UCC=ON")
-    if EXPLICIT_ERROR_CHECK:
-        cmd_str.append("-DNVFUSER_EXPLICIT_ERROR_CHECK=ON")
-    if not NO_NINJA:
-        cmd_str.append("-G")
-        cmd_str.append("Ninja")
-    if not NO_TEST:
-        cmd_str.append("-DBUILD_TEST=ON")
-    if not NO_PYTHON:
-        cmd_str.append("-DBUILD_PYTHON=ON")
-        cmd_str.append(f"-DPython_EXECUTABLE={sys.executable}")
-    if not NO_BENCHMARK:
-        cmd_str.append("-DBUILD_NVFUSER_BENCHMARK=ON")
-    if BUILD_WITH_ASAN:
-        cmd_str.append("-DNVFUSER_BUILD_WITH_ASAN=ON")
-    if BUILD_WITHOUT_DISTRIBUTED:
-        cmd_str.append("-DNVFUSER_DISTRIBUTED=OFF")
-    if BUILD_WITH_SYSTEM_NVTX:
-        cmd_str.append("-DUSE_SYSTEM_NVTX=ON")
-    cmd_str.append(".")
-
-    print(f"Configuring CMake with {' '.join(cmd_str)}")
-    subprocess.check_call(cmd_str)
-
-    max_jobs = multiprocessing.cpu_count()
-    mem_gb_per_task = 3  # Currently compilation of nvFuser souce code takes ~3GB of memory per task, we should adjust this value if it changes in the future.
-    available_mem = get_available_memory_gb()
-    if available_mem > 0:
-        max_jobs_mem = int(available_mem / mem_gb_per_task)
-        max_jobs = min(max_jobs, max_jobs_mem)
-
-    if not CMAKE_ONLY:
-        # build binary
-        max_jobs = os.getenv("MAX_JOBS", str(max_jobs))
-        print(f"Using {max_jobs} jobs for compilation")
-        cmd_str = [
-            get_cmake_bin(),
-            "--build",
-            cmake_build_dir,
-            "--target",
-            "install",
-            "--",
-            "-j",
-            max_jobs,
-        ]
-        subprocess.check_call(cmd_str)
-
-
 def main():
     # NOTE(crcrpar): Deliberately build basically two dynamic libraries here so that they can
     # be treated as "nvfuser_package_data". This function call will put the two of "nvfuser" and
     # "nvfuser_codegen" into "./nvfuser/lib", and the former will be "nvfuser._C".
-    if BUILD_SETUP:
-        cmake()
-    if not CMAKE_ONLY:
+    if config.build_setup:
```

Comment:
my main complain here is, can we merge this function with the one in `python/setup.py`?

I understand we wanted the file here for BC. But I'm nitpicking on how much duplicated logic is kept here.

Reply:
Created `run` in `python/utils.py` and removed the duplicate logic.